### PR TITLE
Use stages to simplify conditions and dependencies between jobs

### PR DESCRIPTION
**Pull request by dannyvv**
This PR leverages Azure Devops Stages.

This cleans up the dependencies and conditions as individual jobs don't have to depend on 'Setup' and have the extra condition.
The only ugly part is having to split out the CLI Init as we want to run this in parralel witht he Universal test builds... And there are no intra-stage dependencies. So this was a balance, when we clean up 'Desktop' we can consider creating separate stages for this.

image

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/microsoft/react-native-windows/pull/6698)

### CODE REVIEW

File: .ado/jobs/cli-init.yml [hunk 1]
```diff
@@ -4,273 +4,265 @@ parameters:
     default : PullRequest
```
Comment (dannyvv) [inline .ado/jobs/cli-init.yml:4]: The only changes here are:
* Extra enum value in the parameter
* An extra `${{ if` above each job to separate the nuget from non nuget
* Removal of the dependsOn and Conditions of the jobs

GitHub's diff is doing a pretty poor job dealing with the indentation...
Opening the PR in codeflow shows a way better diff.

**Comment by dannyvv** [comment]
This change changes the names shown in github a bit... it is now:
PR (<StageName> <JobName> <MatrixEntry>)

I used emoji in the stage name, to split things up a bit. I like it personally, but am open to suggestions. If you have your favorite "professional" emoji for one of the steps I'm happy to update.

**Comment by dannyvv** [comment]
Also if you have a suggestion for splitting some other things to separte stages (like jschecks) let me know. Easy to do :)

**Review by vmoroz — APPROVED** [review]
:shipit:

**Pull request merged**